### ISSUE TITLE
[returners] salt-run jobs.list_jobs fails when using master_job_cache: pgjsonb

### ISSUE THREAD

**Issue by arthurlogilab** [saltstack/salt#32352]
### Description of Issue/Question

```
# salt-run jobs.list_jobs
Exception occurred in runner jobs.list_jobs: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/client/mixins.py", line 342, in low
    data['return'] = self.functions[fun](*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/runners/jobs.py", line 390, in list_jobs
    mret[item] = ret[item]
TypeError: list indices must be integers, not str

```

### Setup

configure master with ```master_job_cache: pgjsonb``` and corresponding postgresql connexion information. 

### Versions Report

```
# salt-run --versions-report
Salt Version:
           Salt: 2015.8.8.2
 
Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: Not Installed
           Mako: 1.0.0
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: 2.3.0
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
 
System Versions:
           dist: debian 8.3 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.3 
 
```

**Comment by jfindlay** [comment]
@arthurlogilab, thanks for reporting.

**Comment by xopher-mc** [comment]
Hey this already is fixed in upstream.  Grab the upstream version of the jsonb returner. 

https://github.com/saltstack/salt/commit/77a3a1470def6818a68a05bc850931c37f1d097d

**Comment by rallytime** [comment]
This fix will also be available in the upcoming release of 2016.3.0.
